wallet-new/integration/Error.hs

```diff
@@ -60,7 +62,8 @@ showConstr = \case
     InvalidTransactionState {} -> "InvalidTransactionState"
     InvalidTransactionFee {} -> "InvalidTransactionFee"
     UnexpectedChangeAddress {} -> "UnexpectedChangeAddress"
-    UnexpectedAccountBalance {} -> "UnexpectedAccountBalance"
+    UnexpectedAccountBalance explanation _ _ ->
+        T.unpack ("UnexpectedAccountBalance " <> explanation)
```

Comment:
This function should not be showing the explanation -- if we need to render that info then it should be in a different function call imo

Reply:
The issue here is that `UnexpectedAccountBalance` is used to describe 2 different unexpected cases, so that which one failed cannot be determined by looking at the test logs.

I'd be happy to remove this change, since it's not strictly related to the rest of the patch. I put it in so that I could see, in the failing test case that apparently only sometimes happens in wallet-new integration tests (seems the actual tests which are run are also chosen randomly), which unexpected balance happened: was the payer not deducted the transfer amount, or was the payee not granted the transfer amount?